clients/pkg/promtail/targets/kafka/target.go

```diff
@@ -80,13 +89,28 @@ func (t *Target) run() {
 		if len(lbs) > 0 {
 			out = out.Merge(lbs)
 		}
-		t.client.Chan() <- api.Entry{
-			Entry: logproto.Entry{
-				Line:      string(message.Value),
-				Timestamp: timestamp(t.useIncomingTimestamp, message.Timestamp),
-			},
-			Labels: out,
+
+		logLines := []string{string(message.Value)}
+
+		if t.messageParser != nil {
+			ll, err := t.messageParser(message)
```

Comment:
rather than nil-checking, I think is best to always inject a message parser, but in the Kafka target case inject a "no-op" implementation, that just returns the sarama message value

Reply:
👍